test/listvolume_tester.py

```diff
@@ -0,0 +1,40 @@
+# import mock
+import test.fake_3par_data as data
+from hpedockerplugin import exception as hpe_exc
+import test.hpe_docker_unit_test as hpedockerunittest
+from hpe3parclient import exceptions
+from oslo_config import cfg
+CONF = cfg.CONF
+
+
+class ListVolumeUnitTest(hpedockerunittest.HpeDockerUnitTestExecutor):
+    def _get_plugin_api(self):
+        return 'volumedriver_list'
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
+
+    def override_configuration(self, config):
+        pass
+
+    # TODO: check_response and setup_mock_objects can be implemented
+    # here for the normal happy path TCs here as they are same
+
+
+class TestListVolumeDefault(ListVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": '', 'Volumes':[]})
+
+        # Check if these functions were actually invoked
+        # in the flow or not
+        mock_3parclient = self.mock_objects['mock_3parclient']
+        mock_3parclient.getWsApiVersion.assert_called()
+

```

Comment:
There are going to be ETCD calls like get_all_vols() and get_path_info_from_vol() during the flow execution. Assert can be made to ensure that those calls were indeed called.

Reply:
Done